src/idl_gen_cpp.cpp

```diff
@@ -1706,6 +1708,158 @@ class CppGenerator : public BaseGenerator {
     code_ += "  }";
   }
 
+bool NeedsCopyCtorAssignOp(const StructDef &struct_def) {
+    for (auto it = struct_def.fields.vec.begin();
+         it != struct_def.fields.vec.end(); ++it) {
+      const auto &field = **it;
+      const auto &type = field.value.type;
+      if (field.deprecated || type.base_type == BASE_TYPE_UTYPE) continue;
+      if (type.base_type == BASE_TYPE_STRUCT) {
+        const auto cpp_type = field.attributes.Lookup("cpp_type");
```

Comment:
all this checking for cpp_type etc seems to just serve to avoid having a copy constructor if a table contains a naked pointer and not other pointers? That doesn't seem worth the complexity.

Reply:
I erred on the side of complexity-in-the-generator instead of suboptimal generated code. I have no problem changing the balance here, especially if tables with cpp_ptr_type == naked are uncommon. Let me know if you want me to clean up here.